kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/OperatorChangeDetectionST.java

```diff
@@ -147,8 +143,70 @@ void shouldRolloutDeploymentWhenFilterConfigurationChanges(String namespace) {
         LOGGER.info("Kafka proxy filter updated");
 
         // Then
+        assertDeploymentUpdated(namespace, kubeClient, originalChecksum);
+    }
+
+    @Test
+    void shouldUpdateDeploymentWhenKafkaProxyChanges(String namespace) {
+        // Given
+        KubeClient kubeClient = kubeClient(namespace);
+        kroxylicious.deployPortIdentifiesNodeWithNoFilters(kafkaClusterName);
+
+        String originalChecksum = getInitialChecksum(namespace, kubeClient);
+
+        KafkaProxy kafkaProxy = kubeClient.getClient().resources(KafkaProxy.class).inNamespace(namespace)
+                .withName(Constants.KROXYLICIOUS_PROXY_SIMPLE_NAME).get();
+
+        // @formatter:off
+        KafkaProxyBuilder updatedKafkaProxy = kafkaProxy.edit()
+                .editOrNewSpec()
+                .withPodTemplate(new PodTemplateSpecBuilder().build())
+                .endSpec();
+
+        // @formatter:on
+
+        // When
+        resourceManager.createOrUpdateResourceWithWait(updatedKafkaProxy);
+        LOGGER.info("Kafka proxy updated");
+
+        // Then
+        assertDeploymentUpdated(namespace, kubeClient, originalChecksum);
+    }
+
+    @Test
+    void shouldUpdateDeploymentWhenSecretChanges(String namespace) {
+        // Given
+        resourceManager.createOrUpdateResourceWithWait(
+                new SecretBuilder().withNewMetadata().withName("upstream-tls-cert").withNamespace(namespace).endMetadata().withType("kubernetes.io/tls")
```

Comment:
the name of the secret is misleading.  it is nothing to do with TLS. It is just any old secret will do.  Let use  something else `turnip`, `kiwi`...